### PR TITLE
Improves pulse guns, ERT, and centcomm areas

### PR DESCRIPTION
**Pull request by ghost**
**Mapping changes:**

ss 2015-02-02 at 03 32 38
- Made the ERT areas smaller, cut out a lot of filler equipment and rooms that were mostly pointless
- Removed the mechs, admins can always spawn these
- Redesigned the ERT area to make better use of lockers and stuff
- Added a backdoor to the admin shuttle so the ERT doesn't get bottlenecked.

**Sprite changes:**

ss 2015-02-01 at 06 25 04

Pulse rifle now has dark blue energy feeds and heat vents along the barrel.

ss 2015-02-01 at 06 25 36

Pulse carbine now better resembles the pulse rifle.

ss 2015-02-02 at 03 41 23

Pulse pistol is now pew pew as fuck.

**Equipment changes:**

ERT now spawn fully equipped and ready to go. 20 minutes of waiting for ERT to gear up is fucking terrible.
ERT lockers have a lot of extra gear in them now.
Rebalanced pulse weapons. Pulse rifle now has 40 pulse shots, carbines have 25, pistols have 10.
Pulse weapons now charge faster.
Pulse weapons are now completely EMP proof as well for even more maximum overpowered.
Removed all the snowflake ammo types and ammo casings for the pulse carbines and stuff (they charge faster now due to a cell var)
Removed ERT suit slowdown.
Various other little misc changes. I'm too tired to find them, so please comb the files and comment on them if you take issue with something.

### CODE REVIEW

**Comment by TZK13** [comment]
I like it, isn't it lacking a room for the ERT commanding officer though?

**Comment by Steelpoint** [comment]
I would rather we keep the Pulse Pistol as a lot of people have liked it.

Also where is the Commanders room, and what items have you removed?

**Comment by paprka** [comment]
We already have a pulse pistol, the pulse 1911. The carbine serves the same exact purpose anyway.

**Comment by Steelpoint** [comment]
I meant the PP sprite. I'll do a proper look at this soon.

**Comment by Steelpoint** [comment]
Ok, complaints about equipment as follows.

EVERYONE SPAWNS WITH A PULSE CARBINE!

Engi
* Not enouch plasteel, should be two stacks. 
* Only one Stun Baton//Magboots/Mesons

Sec
* Only 1 Telescopic Shield/SecBelt/NV Goggles
* Only 1 box of Handcuffs/Flashbangs

Med
* Only one of everything.

There's also no Commanders room, meaning he's missing all of his equipment.

I'm going to stop for that. The layout implies there's only enough equipment for one person.

**Comment by Steelpoint** [comment]
Also I disagree with reworking the mech area. Half of those Mechs are unstoppable admin mechs that will never die and it really limits admin options to give the ERT more equipment that is not DS tier.

**Comment by Steelpoint** [comment]
ALSO. The ERT Officers don't spawn with helmets.

**Comment by Steelpoint** [comment]
I like the idea's pap, but this seems poorly thought out and rushed.

**Comment by paprka** [comment]
You mean like your entire ERT feature?

**Comment by paprka** [comment]
Now there's two of everything, the ERT commander got a fucking .357 in your iteration which is awful. He has a pulse 1911 and an esword now which is more than acceptable imo.

**Comment by Steelpoint** [comment]
I still disagree with giving all ERT Officers Pulse Carbines.

The reason why I gave all Officers a Pulse Pistol, barring the Commander, was to make the Sec Officers more unique as they had access to better Security equipment.

Now what's the point of being a Sec Officer? All your missing are hand cuffs. But also we can better balance out the Pistol.

**Comment by Miauw62** [comment]
Was the 1911 changed? I remember it being a w_class 2 pulse rifle with
infinite ammo.
Op 2-feb.-2015 07:47 schreef "Steelpoint" <notifications@github.com>:

**Comment by paprka** [comment]
Sec officers get a flashbang grenade launcher, telescopic shields, and a pulse rifle

**Comment by Steelpoint** [comment]
Every single balance change, and change, to the ERT is based of my religiously observing and watching the ERT being used in game, as well as going out of my way to get people/admins to give their feedback on the ERT in game and on the forums whenever possible.

Then here you come and do sweeping changes and then go ahead and ignore my feedback.

Here is exactly what I think you need to change.

* Give the Commander all of his prior equipment, this includes making his standard weapon the Autorevolver. The Commander is a thankless job and one of the only perks of it was the cool equipment he had and his gun.
* Give Pulse Pistols the Pulse Pistol sprite. This sprite is my most complimented sprite change as a lot of people really like it. Don't gut it for no reason.
* Only Sec Officers should have Pulse Carbines. Again, what's the point of "balancing" when everyone has the heavy hitting weapon.
* Give the Commander his own office for all his equipment.
* Readd the ERT Mech/Armoury. This is something admins really like as it lets them give the ERT more heavy equipment, as well as spawn stuff there. 

There's undoubtedly more, but that's a start.

I like quite a few of your changes, but I dislike the negatives you made more than the positives.

**Comment by Steelpoint** [comment]
Also, the Pulse Guns are not EMP proof.

**Comment by Steelpoint** [comment]
In case you don't know the Commanders standard equipment, here it is.

* Autorevolver
* (2x) Speed Loaders .357
* Night Vision Security HUD
* Intelicard
* Security Belt
* First Aid Kit
* Box of Handcuffs
* Flash
* Flashbang

A small office will suffice in storing this, or just have him start with it.

**Comment by paprka** [comment]
Steel sometimes you just have to accept that you have a bad vision for the game.

ERT with all these extra frills isn't fun, it's unnecessary and bloated. Admins can still spawn things in the mech bay, but those mechs are for death squads. ERT shouldn't be using mechs anyway, that defeats the point of them having all this gear in the first place. I'm not really opposed to gutting the mechs entirely.

Your pulse pistol sprite was just a modified pulse rifle, it looked like it took you 5 minutes to make. The 1911 has been around forever, replaces the need for the autorevolver since it has more style points, and also replaces the need for the pulse pistol since the carbine can fill that role.

Your game sense is severely lacking steelpoint. Honestly this feature reeks of 'let me spawn a bunch of guns and make a big armory lol!!!' when ERT won't spend a lot of time on centcomm anyway to necessitate such a giant armory and shit.

Why does the ERT commander need:
A NV sec hud
A sec belt
Handcuffs
A first aid kit
Flashbangs

That completely ruins the point of the rest of the officers. Plus, he has access to the rest of the equipment storages anyway, so he can always get gear that the ERT officers are bound to not take.

**Comment by Steelpoint** [comment]
Forget it, I'll just fix things after this is merged.

**Comment by paprka** [comment]
And your petty revert PR will be reverted like they always are. You do realize that this is something cheridan wanted, right? Less bloat with the ERT feature? Less setup time?

**Comment by Steelpoint** [comment]
Thinking I'm going to revert this.

I like the streamlining changes.

I disagree with the equipment changes.

**Comment by paprka** [comment]
We don't need 5000 snowflake weapons. We already have a perfectly fine pulse pistol, why do we need a second one which is a 5 minute sprite that looked like crap anyway?

ss 2015-02-01 at 11 54 15

**Comment by Steelpoint** [comment]
A sprite that even our art directors liked, as well as a lot of people on the servers.

**Comment by paprka** [comment]
"Do you like the pulse pistol?"

"YES!"

Doesn't mean they like the sprite, it means they like the idea of a pulse rifle in pistol form.

Which we already have.

**Comment by Steelpoint** [comment]
screenshot 2015-02-02 15 57 28

In response to the sprites.

I ask people about EVERTHING, and they really like the pistol sprite, as well as the whole concept.

**Comment by paprka** [comment]
>nienhaus
>good opinion on sprites

**Comment by Steelpoint** [comment]
If people were saying the sprites were bad I would have reverted it a long time ago, but people tell me they like it so I've kept it.

**Comment by RemieRichards** [comment]
Locked as per request.

**Comment by Cheridan** [comment]
Unlocked as per request

**Comment by paprka** [comment]
"Merged as per request"

No but OP updated.

**Comment by Steelpoint** [comment]
I'll do a one over but I'm pretty happy with this.

**Comment by paprka** [comment]
You can always change the contents of the lockers and stuff later steelpoint, the only stuff here that's important to me is that extra gear is mostly in lockers and everything else is spawned in. Also, I fixed the issue stickymayhem had with the door not having any access requirements.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -122,6 +122,8 @@
 		overheat = 0
 		recent_reload = 0
 	..()
+/obj/item/weapon/gun/energy/kinetic_accelerator/emp_act(severity)
+	return
```
Comment (paprka) [inline code/modules/projectiles/guns/energy/special.dm]: This is a bandaid over an earlier fix to ebows and KAs breaking when EMP'd. They are lever-action, and EMP shouldn't affect them.

**Comment by Steelpoint** [comment]
Correct me if I'm wrong but the Med Officers don't get the Nanite Combat Injector?

File: code/game/objects/items/weapons/power_cells.dm [hunk 2]
```diff
@@ -103,16 +122,15 @@
...
 	rating = 1
 
-/obj/item/weapon/stock_parts/cell/slime
+/obj/item/weapon/stock_parts/cell/high/slime
 	name = "charged slime core"
 	desc = "A yellow slime core infused with plasma, it crackles with power."
 	origin_tech = "powerstorage=2;biotech=4"
-	icon = 'icons/mob/slimes.dmi' //'icons/obj/harvest.dmi'
-	icon_state = "yellow slime extract" //"potato_battery"
-	maxcharge = 10000
```
Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/power_cells.dm]: Why exactly are xenobio's slime cells being reduced by a factor of ten? Is this a balance issue somehow? I fail to see how limited production of High-capacity powercells is so bad.

Comment (paprka) [inline code/game/objects/items/weapons/power_cells.dm]: This has the same exact charge as it did before, it's just snowflake removal. High cap cells have 10000.

**Comment by paprka** [comment]
^ @Steelpoint

**Comment by Steelpoint** [comment]
Love you too.

**Comment by Deantwo** [comment]
Now it is more like a meeting room, I like it.

File: code/game/objects/items/weapons/power_cells.dm (hunk 2 above)
Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/power_cells.dm]: Didn't spot that, please ignore.

**Comment by Alek2ander** [comment]
Why does ERT need C4?

**Comment by Stickymayhem** [comment]
Thanks pap this is great?

What does each locker come with? A full set of equipment? It would be extremely helpful for admin events if I could just drop a locker with everything but the space suits ready inside for each class.

**Comment by Miauw62** [comment]
You shouldn't be balancing the ERT tbh, they're an admin event. Give them a
few lockers with SOME basic equipment and let the admins spawn in the rest
at their own discretion.
Op 2-feb.-2015 20:52 schreef "Stickymayhem" <notifications@github.com>:

**Comment by Steelpoint** [comment]
It takes a lot more effort for admins to spawn in individual items that it takes to have those items prespawned.

**Comment by paprka** [comment]
@Stickymayhem You don't even need to do that, you can just select equipment > emergency response officer > then select the class you want to dress them up as with basic gear if you want to select individual ckeys instead of having them random-selected.

**Comment by paprka** [comment]
Conflicts fixed here. This changes a lot of stuff and I would prefer it be expedited if someone has a chance to look at it so steelpoint can make equipment changes to spawning/lockers if he feels it's necessary.

**Comment by AnonymousNow** [comment]
If ERT now spawn fully equipped, does this mean it assigns officer types? Because ideally it should give you a choice on spawning.

**Comment by paprka** [comment]
No, it assigns them at random, I'm not gonna go through the code mess that is allowing people to choose.

**Comment by AnonymousNow** [comment]
That's really rather unfortunate, seeing as how this draws on a pool of existing player characters, and it would make more sense for some to be X or Y job representative.

**Comment by Steelpoint** [comment]
I just noticed but you forgot to give the Ion Guns firing pins.

**Comment by paprka** [comment]
rip

**Pull request merged**